File: src/seeder/address.cpp
// Copyright (c) 2017 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

#include "address.h"

CSeederAddress::CSeederAddress() : CSeederService() {
    Init();
}

CSeederAddress::CSeederAddress(CSeederService ipIn, uint64_t nServicesIn)
    : CSeederService(ipIn) {
    Init();
    nServices = nServicesIn;
}

void CSeederAddress::Init() {
    nServices = NODE_NETWORK | NODE_BITCOIN_CASH;
    nTime = 100000000;
}

void CSeederAddress::print() const {
    printf("CSeederAddress(%s)\n", ToString().c_str());
}

File: modules/ecash-parse/src/address.ts
// Copyright (c) 2026 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

export const previewAddress = (address: string): string => {
    const addressWithoutPrefix = address.includes(':')
        ? address.split(':')[1]
        : address;

    return `${addressWithoutPrefix.slice(0, 3)}...${addressWithoutPrefix.slice(
        -3,
    )}`;
};
